api/v1/mdb/mongodb_types.go

```diff
@@ -801,6 +807,22 @@ func (s *Security) IsTLSEnabled() bool {
 	return s.CertificatesSecretsPrefix != ""
 }
 
+func (s *Security) IsOIDCEnabled() bool {
```

Comment:
nit, also just an opinion:

```suggestion
func (s *Security) IsOIDCEnabled() bool {
	if s == nil || s.Authentication == nil || !s.Authentication.Enabled {
		return false
	}
	return s.Authentication.IsOIDCEnabled()
}
```